### PR TITLE
BLOCKED BY SG: Add support for argument availability_zone_name

### PR DESCRIPTION
**Pull request by auebergang**
## what
* Adds support for argument `availability_zone_name`

## why
* `availability_zone_name` creates the filesystem as a One Zone storage class

## references
* [Resource: aws_efs_file_system](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/efs_file_system)

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -6,6 +6,7 @@ locals {
 resource "aws_efs_file_system" "default" {
   count                           = module.this.enabled ? 1 : 0
   tags                            = module.this.tags
+  availability_zone_name          = var.availability_zone_name
```
[the remark this answers is not in the document]
Comment (Gowiem) [reply main.tf:9]: @auebergang can you do us a favor and add the following above line 7?

```
#bridgecrew:skip=BC_AWS_GENERAL_48 - Skipping as backup plan can be added via https://github.com/cloudposse/terraform-aws-backup
```

See here for an example: https://github.com/cloudposse/terraform-aws-alb/blob/master/main.tf#L140

**Comment by Gowiem** [comment]
@auebergang This looks good and I'm happy to get it merged, but we are hitting two issues: 

1. The security issue with BridgeCrew. See the above comment inline on BC's comment. 
2. Auto format is failing -- Try running the following locally: 

```bash
make init
make readme
git add . 
git commit -m "chore: readme updates"
git push
```

**Comment by auebergang** [comment]
Updated. I've added an update to the var description as well to note providing a single subnet in the same AZ.

**Comment by nitrocode** [comment]
Thanks for your contribution! This has been released as version [0.32.0](https://github.com/cloudposse/terraform-aws-efs/releases/tag/0.32.0).

**Pull request closed without merging**